### PR TITLE
operator cleanup

### PR DESCRIPTION
**Pull request by yanggangtony**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Do some operator cleanup.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
None
```

### CODE REVIEW

File: operator/pkg/deinit.go [hunk 1]
```diff
@@ -103,12 +103,12 @@ func defaultJobDeInitOptions() *DeInitOptions {
 
 // NewDeInitOptWithKarmada returns a DeInitOpt function to initialize DeInitOptions with karmada resource
 func NewDeInitOptWithKarmada(karmada *operatorv1alpha1.Karmada) DeInitOpt {
-	return func(opt *DeInitOptions) {
```
[the remark this answers is not in the document]
Comment (yanggangtony) [reply operator/pkg/deinit.go:106]: 😂😂
Both is good. 
But now the same function method , the args is not the same .
one is opt and the other is options.

[the remark this answers is not in the document]
Comment (calvin0327) [reply operator/pkg/deinit.go:106]: I'm with you on arg name`options`, please update other functions with `opt` args.  like:

https://github.com/karmada-io/karmada/blob/master/operator/pkg/deinit.go#L39
https://github.com/karmada-io/karmada/blob/master/operator/pkg/init.go#L66

[the remark this answers is not in the document]
Comment (yanggangtony) [reply operator/pkg/deinit.go:106]: How about also update the 
[https://github.com/karmada-io/karmada/blob/218a0b2400958e6e13d1044e2f6de6fcb30dd867/operator/pkg/init.go#L40](https://github.com/karmada-io/karmada/blob/218a0b2400958e6e13d1044e2f6de6fcb30dd867/operator/pkg/init.go#L40).

And the [https://github.com/karmada-io/karmada/blob/218a0b2400958e6e13d1044e2f6de6fcb30dd867/operator/pkg/deinit.go#L25](https://github.com/karmada-io/karmada/blob/218a0b2400958e6e13d1044e2f6de6fcb30dd867/operator/pkg/deinit.go#L25)

[the remark this answers is not in the document]
Comment (yanggangtony) [reply operator/pkg/deinit.go:106]: Rebased..

File: operator/pkg/deinit.go [hunk 2]
```diff
@@ -84,13 +84,13 @@ func (data *deInitData) RemoteClient() clientset.Interface {
 
 // NewJobDeInitOptions calls all of DeInitOpt func to initialize a DeInitOptions.
 // if there is not DeInitOpt functions, it will return a default DeInitOptions.
-func NewJobDeInitOptions(opts ...DeInitOpt) *DeInitOptions {
-	options := defaultJobDeInitOptions()
+func NewJobDeInitOptions(options ...DeInitOpt) *DeInitOptions {
```
Comment (calvin0327) [inline operator/pkg/deinit.go:87]: DeInitOpt is different with DeInitOptions, I think `opts` is more appropriate here. 
What do you think?

Comment (yanggangtony) [reply operator/pkg/deinit.go]: Yes,  its ok

File: operator/pkg/deinit.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Poor12) [reply operator/pkg/deinit.go:106]: About options, we have discussed in https://github.com/karmada-io/karmada/pull/3398. Using `o` is more clearer.

[the remark this answers is not in the document]
Comment (yanggangtony) [reply operator/pkg/deinit.go:106]: ```o``` is more simple and brevity.
But how to represents the ```Singular or Plural```

[the remark this answers is not in the document]
Comment (Poor12) [reply operator/pkg/deinit.go:106]: Options will always be plural, I think.

**Comment by yanggangtony** [comment]
😂😂
According @Poor12 's opinion.
And these similar question have discussed in [https://github.com/karmada-io/karmada/pull/3398](https://github.com/karmada-io/karmada/pull/3398).

 Using o is more clearer.

So may i changes the ```options *DeInitOptions``` to ```opt *DeInitOptions```?

**Comment by calvin0327** [comment]
> 😂😂 According @Poor12 's opinion. And these similar question have discussed in #3398.
> 
> Using o is more clearer.
> 
> So may i changes the `options *DeInitOptions` to `opt *DeInitOptions`?

To be consistent with other components, we can use `o` as the arg name.
`options *DeInitOptions`  ->  `o *DeInitOptions`,  I think it's ok.

**Comment by yanggangtony** [comment]
rebased.

**Pull request merged**